remote_attestation/java/src/com/google/oak/remote_attestation/Message.java

```diff
@@ -0,0 +1,356 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package com.google.oak.remote_attestation;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInput;
+import java.io.DataInputStream;
+import java.io.DataOutput;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import static java.lang.Math.toIntExact;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/** Wrapper class for Remote Attestation Protocol messages. */
+public class Message {
+  static final int CLIENT_HELLO_HEADER = 1;
+  static final int SERVER_IDENTITY_HEADER = 2;
+  static final int CLIENT_IDENTITY_HEADER = 3;
+
+  /** Remote Attestation protocol version. */
+  public static final int PROTOCOL_VERSION = 1;
+  /**
+   * Size (in bytes) of the prefix that describes the size of a serialized array.
+   * Prefix is encoded with Little-Endian.
+   */
+  public static final int SERIALIZATION_PREFIX_SIZE_BYTES = 8;
+
+  /** Size (in bytes) of a random array sent in messages to prevent replay attacks. */
+  public static final int REPLAY_PROTECTION_ARRAY_SIZE_BYTES = 32;
+  public static final int EPHEMERAL_PUBLIC_KEY_SIZE_BYTES = 32;
+  // TODO(#2277): Use OpenSSL signature format (which is 72 bytes).
+  public static final int TRANSCRIPT_SIGNATURE_SIZE_BYTES = 64;
+  public static final int SIGNING_PUBLIC_KEY_SIZE_BYTES = 65;
+
+  /** Initial message that starts Remote Attestation handshake. */
+  public static class ClientHello {
+    /** Message header. */
+    private final byte header;
+    /** Random vector sent in messages for preventing replay attacks. */
+    private final byte[] random;
+
+    public ClientHello(byte[] random) {
```

Comment:
Doesn't this need to be of a fixed size? It seems here nothing is checking the size.

Reply:
It checks the size once it tries to serialize.
I deleted the check in the conscructor, so that it doesn't throw exceptions.